pkg/gofr/datasource/file/ftp/interface.go

```diff
@@ -3,11 +3,82 @@ package ftp
 import (
 	"context"
 	"io"
+	"os"
 	"time"
 
 	"github.com/jlaffaye/ftp"
 )
 
+// FTPFile represents a file in the filesystem.
+type FTPFile interface {
+	io.Closer
+	io.Reader
+	io.ReaderAt
+	io.Seeker
+	io.Writer
+	io.WriterAt
+
+	ReadAll() (RowReader, error)
+}
+
+type FileInfo interface {
+	Name() string       // base name of the file
+	Size() int64        // length in bytes for regular files; system-dependent for others
+	ModTime() time.Time // modification time
+	Mode() os.FileMode  // file mode bits
+	IsDir() bool        // abbreviation for Mode().IsDir()
+}
+
+type RowReader interface {
+	Next() bool
+	Scan(interface{}) error
+}
+
+// FTPFileSystem : Any simulated or real filesystem should implement this interface.
+type FTPFileSystem interface {
```

Comment:
```suggestion
// FileSystem : Any simulated or real filesystem should implement this interface.
type FileSystem interface {
```

Reply:
`FileSystem` is also a struct so we need to rename it to `FTPFileSystem`